ui/src/components/dialog/__tests__/QDialog.cy.js

```diff
@@ -26,103 +136,425 @@ describe('Dialog API', () => {
       })
 
       describe('(prop): auto-close', () => {
-        it.skip(' ', () => {
-          //
+        it('should auto-close the dialog', () => {
+          const model = ref(true)
+          mountQDialogWrapper({
+            props: {
+              ...vModelAdapter(model),
+              autoClose: true
+            }
+          })
+
+          cy.withinDialog(() => {
+            cy.dataCy('dialog-button').click()
+          })
         })
       })
 
       describe('(prop): no-refocus', () => {
-        it.skip(' ', () => {
-          //
+        it('should not refocus on the DOM element that had focus', () => {
+          const model = ref(false)
+
+          mountQDialogWrapper({
+            props: {
+              ...vModelAdapter(model),
+              noRefocus: false
+            }
+          })
+
+          // Test the input field regaining focus
+          cy.dataCy('input-field')
+            .focus()
+          cy.dataCy('input-field')
+            .then(async () => {
+              model.value = true
+              await nextTick()
```

Comment:
I'm not sure `nextTick` is actually required when changing the refs, to make sure there's no race condition, check it out / test it / search for confirmation about this (I guess on VTU repo)

Reply:
In this particular case, `nextTick` is required. According to VTU docs, if we await `setProps`, `setValue`, etc then it is equivalent to having called the `await Vue.nextTick()` after the command.  I noticed that the `vModelAdapter` implementation does not await when calling `Cypress.vueWrapper.setProps({ [modelName]: value })`. So in this case we need `nextTick` to ensure the view has updated.

Reply:
Shall we update `vModelAdapter` to await for `setProps` then?
I'm not sure the interaction would work as expected, due to the use of reactivity we do in tests, but we could try that

Reply:
Go on and test this

Reply:
Tested the edit and it does not break any tests. I have opened a PR [here](https://github.com/quasarframework/quasar-testing/pull/364)